src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -206,18 +208,17 @@ public static ClassName get(TypeElement element) {
     checkNotNull(element, "element == null");
     String simpleName = element.getSimpleName().toString();
 
-    switch (element.getNestingKind()) {
-      case TOP_LEVEL:
-        Name packageName = ((PackageElement) element.getEnclosingElement()).getQualifiedName();
-        return new ClassName(packageName.toString(), null, simpleName);
-
-      case MEMBER:
-        ClassName enclosingClass = get((TypeElement) element.getEnclosingElement());
-        return enclosingClass.nestedClass(simpleName);
-
-      default:
-        throw new IllegalArgumentException("unexpected type nesting");
+    Element enclosingElement = element.getEnclosingElement();
+    if (enclosingElement instanceof PackageElement) {
```

Comment:
Any way to do this without `instanceof`? My read of the docs is that a single class is allowed to implement multiple types.